ssr/src/page/post_view/mod.rs

```diff
@@ -278,6 +277,134 @@ pub fn PostViewWithUpdates(initial_post: Option<PostDetails>) -> impl IntoView {
             fetch_next_videos=next_videos
             queue_end
             overlay=|| view! { <HomeButtonOverlay /> }
+            threshold_trigger_fetch=10
+        />
+    }
+}
+
+#[component]
+pub fn PostViewWithUpdatesMLFeed(initial_post: Option<PostDetails>) -> impl IntoView {
+    let PostViewCtx {
+        fetch_cursor,
+        video_queue,
+        current_idx,
+        queue_end,
+    } = expect_context();
+
+    let recovering_state = create_rw_signal(false);
+    if let Some(initial_post) = initial_post.clone() {
+        fetch_cursor.update_untracked(|f| {
+            // we've already fetched the first posts
+            if f.start > 1 || queue_end.get_untracked() {
+                recovering_state.set(true);
+                return;
+            }
+            f.start = 1;
+            f.limit = 15;
+        });
+        video_queue.update_untracked(|v| {
+            if v.len() > 1 {
+                // Safe to do a GC here
+                let rem = 0..(current_idx.get_untracked().saturating_sub(6));
+                current_idx.update(|c| *c -= rem.len());
+                v.drain(rem);
+                return;
+            }
+            *v = vec![initial_post];
+        })
+    }
+    let (nsfw_enabled, _, _) = use_local_storage::<bool, FromToStringCodec>(NSFW_TOGGLE_STORE);
+    let auth_canisters: RwSignal<Option<Canisters<true>>> = expect_context();
+
+    let fetch_video_action = create_action(move |_| async move {
```

Comment:
Since only this action is different, lets try to generalize PostView so code duplication is minimal?